In [10]:
import pandas as pd

## Data

In [11]:
df = pd.read_csv('EjemploDatos.csv')
print('Nº of rows: ', df.shape[0])
df.head()

Nº of rows:  9


prediction date  10u_(44.0, -9.5)  10u_(44.0, -9.375)  \
0  2016-01-01 00:00:00          1.577097            1.675974   
1  2016-01-01 01:00:00          1.320722            1.059857   
2  2016-01-01 02:00:00          1.562336            1.023518   
3  2016-01-01 03:00:00          0.198002            0.025639   
4  2016-01-01 04:00:00          1.604583            1.043792   

   10u_(44.0, -9.25)  10u_(44.0, -9.125)  10u_(44.0, -9.0)  \
0           1.836863            1.821115          1.681467   
1           1.200238            1.276654          1.713910   
2           0.415607            0.335896          0.563801   
3           0.332890           -0.118526         -0.533199   
4           1.073821            0.065032         -0.961823   

   10u_(44.0, -8.875)  10u_(44.0, -8.75)  10u_(44.0, -8.625)  \
0            1.424021           1.190012            1.106394   
1            1.246137           0.725141            0.203412   
2            0.696369           0.487873            0.106770   
3           -1.093990          -1.123653           -1.040646   
4           -1.394685          -1.394074           -1.230012   

   10u_(44.0, -8.5)  ...  vel100_(42.875, -6.0)  vel100_(43.0, -6.0)  \
0          1.108713  ...               4.542353             5.321404   
1          0.215253  ...               4.708214             5.435709   
2         -0.157024  ...               4.553245             5.325067   
3         -0.928341  ...               4.668390             5.177444   
4         -1.500520  ...               4.791137             5.317643   

   vel100_(43.125, -6.0)  vel100_(43.25, -6.0)  vel100_(43.375, -6.0)  \
0               5.971150              5.862873               5.164530   
1               6.092992              5.361253               3.485183   
2               6.018884              4.991258               2.281320   
3               5.716941              4.493564               1.185751   
4               5.648662              4.902998               3.442938   

   vel100_(43.5, -6.0)  vel100_(43.625, -6.0)  vel100_(43.75, -6.0)  \
0             5.391425               6.107658              6.398007   
1             3.988883               4.845162              5.260955   
2             2.807106               3.654994              4.170403   
3             1.837592               3.329041              4.260316   
4             3.724846               4.111763              4.328406   

   vel100_(43.875, -6.0)  vel100_(44.0, -6.0)  
0               6.472929             6.793221  
1               5.464975             5.908251  
2               4.632105             5.078648  
3               4.934864             5.643699  
4               4.885375             5.730442  

[5 rows x 3481 columns]

The EjemploDatos.csv contains 3481 columns that can be divided into: 
 - Prediction Date : day and hour 
 
 - 10u             : u component at height 10m
  
 - 10v             : v component at height 10m
 
 - 2t              : temperature at height 2m
 
 - sp              : pressure at surface
 
 - 100u            : u component at height 100m
 
 - 100v            : v component at height 100m
 
 - vel10           : module of the wind at height 10m
 
 - vel100          : module of the wind at height 100m
 

## Target

Observed production at Prediction Date (MWh)

Potencia máxima del parque: 17560Kwh (los datos, ya confirmado, están en Kwh

In [12]:
target_df = pd.read_csv('EjemploTarget.csv',
                        header=None, 
                        names=['Prediction Date','Target (KWh)'])
target_df.head()

Prediction Date  Target (KWh)
0  2016-01-01 00:00:00       4126.70
1  2016-01-01 01:00:00       5688.51
2  2016-01-01 02:00:00       7742.01
3  2016-01-01 03:00:00       7660.50
4  2016-01-01 04:00:00       9730.48

### Coordinates of interest

![Map](./figures/mapa.png)

In [13]:
# coordinates of interest
coordenates_of_interest = ['prediction date',

                            '(43.875, -8.375)', '(43.75, -8.375)', '(43.625, -8.375)', '(43.5, -8.375)', '(43.375, -8.375)', '(43.25, -8.375)', '(43.125, -8.375)', '(43.0, -8.375)', '(42.875, -8.375)',

                            '(43.875, -8.25)', '(43.75, -8.25)', '(43.625, -8.25)', '(43.5, -8.25)', '(43.375, -8.25)', '(43.25, -8.25)', '(43.125, -8.25)', '(43.0, -8.25)', '(42.875, -8.25)',
                            
                            '(43.875, -8.125)', '(43.75, -8.125)', '(43.625, -8.125)', '(43.5, -8.125)', '(43.375, -8.125)', '(43.25, -8.125)', '(43.125, -8.125)', '(43.0, -8.125)', '(42.875, -8.125)',

                           '(43.875, -8.0)', '(43.75, -8.0)', '(43.625, -8.0)', '(43.5, -8.0)', '(43.375, -8.0)', '(43.25, -8.0)', '(43.125, -8.0)', '(43.0, -8.0)', '(42.875, -8.0)',
                        
                           '(43.875, -7.875)','(43.75, -7.875)','(43.625, -7.875)','(43.5, -7.875)','(43.375, -7.875)','(43.25, -7.875)','(43.125, -7.875)','(43.0, -7.875)','(42.875, -7.875)',

                            '(43.875, -7.75)','(43.75, -7.75)','(43.625, -7.75)','(43.5, -7.75)','(43.375, -7.75)','(43.25, -7.75)','(43.125, -7.75)','(43.0, -7.75)','(42.875, -7.75)',

                            '(43.875, -7.625)','(43.75, -7.625)','(43.625, -7.625)','(43.5, -7.625)','(43.375, -7.625)','(43.25, -7.625)','(43.125, -7.625)','(43.0, -7.625)','(42.875, -7.625)',

                            '(43.875, -7.5)','(43.75, -7.5)','(43.625, -7.5)','(43.5, -7.5)','(43.375, -7.5)','(43.25, -7.5)','(43.125, -7.5)','(43.0, -7.5)','(42.875, -7.5)',

                            '(43.875, -7.375)','(43.75, -7.375)','(43.625, -7.375)','(43.5, -7.375)','(43.375, -7.375)','(43.25, -7.375)','(43.125, -7.375)','(43.0, -7.375)','(42.875, -7.375)',
                            
                           ]

In [14]:
# keep columns that contain any of the coordinates of interest
df = df.loc[:, df.columns.str.contains('|'.join(coordenates_of_interest))]

/var/folders/_z/8tjgjv9d62qfwnfqwr1z6yl00000gq/T/ipykernel_54463/4242011990.py:2: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  df = df.loc[:, df.columns.str.contains('|'.join(coordenates_of_interest))]


The resulting data has the same dimension as the data used in the paper "Deep Learning Applied to Wind Power Forecasting: a Spatio-Temporal Approach" fed to the Vision Transformer. 

In [15]:
df.head()

prediction date  10u_(43.875, -8.375)  10u_(43.875, -8.25)  \
0  2016-01-01 00:00:00              0.646799             0.940744   
1  2016-01-01 01:00:00              0.066449             0.209149   
2  2016-01-01 02:00:00             -0.248577            -0.208050   
3  2016-01-01 03:00:00             -0.939205            -0.826656   
4  2016-01-01 04:00:00             -1.553864            -1.209870   

   10u_(43.875, -8.125)  10u_(43.875, -8.0)  10u_(43.875, -7.875)  \
0              1.023264            0.863474              0.737864   
1              0.326459            0.294477              0.429730   
2             -0.377850           -0.632000             -0.777508   
3             -1.048580           -1.244015             -1.253048   
4             -1.632600           -2.067414             -2.279450   

   10u_(43.875, -7.75)  10u_(43.875, -7.625)  10u_(43.875, -7.5)  \
0             0.503366              0.272409            0.131907   
1             0.099896             -0.319903           -0.740069   
2            -0.984661             -1.295086           -1.629803   
3            -1.262936             -1.560299           -2.039181   
4            -2.524446             -2.592317           -2.750764   

   10u_(43.875, -7.375)  ...  vel100_(43.875, -7.5)  vel100_(42.875, -7.375)  \
0              0.223703  ...               7.685605                 6.777981   
1             -0.581500  ...               7.588559                 6.671898   
2             -1.254192  ...               8.390730                 7.544478   
3             -2.125484  ...               9.077174                 8.745656   
4             -2.681672  ...               8.777578                 8.605917   

   vel100_(43.0, -7.375)  vel100_(43.125, -7.375)  vel100_(43.25, -7.375)  \
0               7.208619                 8.079643                8.283714   
1               7.046983                 8.140311                8.318516   
2               7.833109                 8.566935                8.423622   
3               8.588821                 8.887086                8.570982   
4               8.417324                 8.805204                8.697603   

   vel100_(43.375, -7.375)  vel100_(43.5, -7.375)  vel100_(43.625, -7.375)  \
0                 8.437874               8.417049                 7.833382   
1                 8.475589               8.320119                 7.561365   
2                 8.208681               7.951049                 7.657357   
3                 8.070685               7.524815                 7.277965   
4                 8.174285               7.259660                 6.497139   

   vel100_(43.75, -7.375)  vel100_(43.875, -7.375)  
0                7.427651                 7.626461  
1                7.171789                 7.649203  
2                7.868434                 8.506869  
3                8.109472                 9.125601  
4                7.239294                 8.592414  

[5 rows x 649 columns]